deployment/promotion/promote.py

```diff
@@ -9,396 +9,157 @@
-    return dest_build
-
-
-def promote(_config):
-    promotion = get_promotion_data(_config)
-    promoted_build = create_promotion(_config, promotion)
-    print("Promoted build %d created." % promoted_build['id'])
-
-
-promote(config)
+    print("Fetching or creating destination registered model %s" % source_model.name)
+    dest_model = dest_client.get_or_create_registered_model(name=source_model.name)
+
+# Copy information from the source model to the destination model
+# TODO: add readme, description to the client
+print("Copying info from source model to destination model")
+dest_model.add_labels([v for v in source_model.get_labels()])
+
+# Create the basic model version
+# TODO: add readme, description to the client
+# TODO: check return types
+print("Copying info from source model version to destination model version")
+dest_model_version = dest_model.create_version(
+    name=source_model_version.name,
+    labels=[v for v in source_model_version.get_labels()],
+    attrs={k: v for k, v in source_model_version.get_attributes().items()},
+)
+
+for key in source_model_version.get_artifact_keys():
+    artifact = source_model_version.get_artifact(key)
+    dest_model_version.log_artifact(key, artifact, overwrite=True)
+
+model = source_model_version.get_model()
+dest_model_version.log_model(model, custom_modules=[], overwrite=True)
+
+env = source_model_version.get_environment()
+dest_model_version.log_environment(env, overwrite=True)
+
+print("Copying build information")
+source_builds = source_model_version.list_builds()
+if len(source_builds) == 0:
+    raise ValueError("No builds found for model version %s" % source_model_version_id)
+source_build = source_builds[0]
+
+dest_build = dest_model_version.create_external_build(
+    location=source_build.location,
+    requires_root=source_build.requires_root,
+    scan_external=source_build.scan_external,
+    self_contained=source_build.self_contained,
+)
+dest_build.message = source_build.message
+print(dest_build)
+
+if not source_build.self_contained:
+    print("Got non self contained build. Copying artifacts in S3")
+    source_bucket = os.environ.get('VERTA_SOURCE_S3_BUCKET')
+    if not source_bucket:
+        raise KeyError("Missing environment variable VERTA_SOURCE_S3_BUCKET")
+    dest_bucket = os.environ.get('VERTA_DEST_S3_BUCKET')
+    if not dest_bucket:
+        raise KeyError("Missing environment variable VERTA_DEST_S3_BUCKET")
+
+    import boto3
+
+    keys = source_model_version.get_artifact_keys()
+    keys.append(source_model_version._MODEL_KEY)
+
+    for key in keys:
+        path = source_model_version._get_artifact_msg(key).path
+        print(f"Copying {key} via location {path}")
+
+        s3 = boto3.client('s3')
+
+        # Download the file from the source bucket
+        s3.download_file(source_bucket, path, 'artifact.tmp')
+
+        # Upload the downloaded file to the destination bucket
+        s3.upload_file('artifact.tmp', dest_bucket, path)
```

Comment:
From the docs, it looooooks like `artifact.tmp` will be overwritten by `path`. This warrants a comment, I believe, as it is confusing and required looking it up.

Reply:
I'm not sure what you mean. The file in the local system won't be overwritten in this call.

Reply:
no, the file on s3. I found it confusing that it looks like you're writing a file named `artifact.tmp` to s3 when you're actually not (you're writing `model.pkl` and `model_api.json`, etc). 🤷🏼‍♀️